### DISCUSSION TITLE
Validate a form before adding/patching?

### DISCUSSION THREAD

**Discussion by regisin** [jowilf/sqlalchemy-file#125] [Q&A] [answered]
Hi, I'm using `starlette-admin` with `sqlalchemy-file`, and `fastapi-users` to handle auth.

I extended the User model to have an avatar/profile picture. I am able to update/add/etc from the admin panel provided by `starlette-admin`, as well as from the patch route below.

My `User` model looks like this:
```python
class User(SQLAlchemyBaseUserTableUUID, Base):
    name: str = Column(String)
    avatar = Column(ImageField(upload_storage="avatar"))
```

And I have a dedicated route to update the avatar:
```python
@router.patch('/users/me/profile-picture', response_model=UserRead, tags=["users"])
async def upload_profile_picture(
        file: UploadFile = File(...),
        user: User = Depends(current_active_user),
        manager: UserManager = Depends(get_user_manager)
):
    user = await manager.user_db.update(
        user, {"avatar": file}
    )
    return user
```

Is there a good way to validate the `file` before calling `.update(...)`? The exception does happen, and the validator is triggered, but the feedback the user has is a 500 error. Can I somehow use the validator myself and raise a 400 bad request instead? How?

**Comment by jowilf** [comment, marked as answer]
You can simply wrap your udpdate with a try/catch block https://github.com/jowilf/sqlalchemy-file/blob/fc91c904c483dffc4db5c30f7d77eb77b199f3c8/examples/fastapi/app.py#L117C1-L117C1 or handle the error globally https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions

**Comment by regisin** [comment]
Well, now I feel silly as that sounds like the obvious and elegant way to do it. Thanks!
